kratos/tests/utilities/test_geometrical_projection_utilities.cpp

```diff
@@ -0,0 +1,106 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:     BSD License
+//  			 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+//
+
+// Project includes
+#include "testing/testing.h"
+#include "utilities/mortar_utilities.h"
+
+#include "geometries/triangle_2d_3.h"
+
+namespace Kratos
+{
+namespace Testing
+{
+
+using NodeType = typename MortarUtilities::NodeType;
+using GeometryType = typename MortarUtilities::GeometryType;
+
+GeometryType::Pointer CreateTriangle2D3NForTest()
+{
+    GeometryType::PointsArrayType points;
+    points.push_back(Kratos::make_shared<NodeType>(0.04, 0.02, 0.0));
+    points.push_back(Kratos::make_shared<NodeType>(1.1, 0.03, 0.0));
+    points.push_back(Kratos::make_shared<NodeType>(1.08, 1.0, 0.0));
+
+    return GeometryType::Pointer(new Triangle2D3<NodeType>(points));
```

Comment:
This is a 2D triangle, are you trying to project in 3d space or 2D, in 2D space the geometry shoul be a line, not the whole triangle

Reply:
![imagen](https://user-images.githubusercontent.com/19991680/40911037-0c2f516a-67ee-11e8-9c0f-e21ece0fe174.png)
This is  representation of the geometry

Reply:
This is 2D, the normal you defined is out of the plane, I don't understand

Reply:
copy paste error :)
Should be 3D ofc. However looking into the code I am not sure if it makes a difference, no functions from the Geometry are being used, only the first point

Reply:
I am also trying it with 3D Triangle now

The second test also does not work, and there is no geometry being used there

Reply:
OK, in any case if the Z coords is 0, the expected distance is 0